controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -66,18 +68,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	if err := r.addToolchainClusterRoleLabelFromType(ctx, toolchainCluster); err != nil {
 		return reconcile.Result{}, err
 	}
-
 	return reconcile.Result{}, r.clusterCacheService.AddOrUpdateToolchainCluster(toolchainCluster)
 }
 
 func (r *Reconciler) addToolchainClusterRoleLabelFromType(ctx context.Context, toolchainCluster *toolchainv1alpha1.ToolchainCluster) error {
 	logger := log.FromContext(ctx)
 
-	if clusterType, found := toolchainCluster.Labels[cluster.LabelType]; !found {
-		logger.Info("cluster `type` label not found, unable to add toolchain cluster role label from type")
-		return nil
-	} else if clusterType != string(cluster.Member) {
-		logger.Info("cluster `type` is not member, skipping cluster role label setting")
+	if !r.useClusterRL {
```

Comment:
See my comment above , you could pass `cluster.Tenant` instead of using a bool here, and you could even avoid calling this function `addToolchainClusterRoleLabelFromType` when this `r.setClusterRole` variable is empty.

Also I would rename the function from `addToolchainClusterRoleLabelFromType` to something like `addToolchainClusterRole`.

Reply:
as I mentioned here https://github.com/codeready-toolchain/toolchain-common/pull/359#discussion_r1504144684
we should drop the whole `addToolchainClusterRoleLabelFromType` function. The roles should be set when the ToolchainCluster CR is created https://github.com/codeready-toolchain/toolchain-cicd/blob/34ea85fe21ae05c9bb1a0a2bbdc986b94c8d9e92/scripts/add-cluster.sh#L387-L400 
IIRC, this was a function to migrate the already existing CRs.